### PR TITLE
Run build workflow on tags as well

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it:

Run build workflow on tags as well, otherwise we’ll not be able to run it for releases.

#### Which issue(s) this PR fixes:

None
#### Does this PR have test?
None

#### Special notes for your reviewer:
None
/hold 
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@kubernetes-sigs/security-profiles-operator-maintainers PTAL

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -44,7 +46,7 @@ jobs:
     if: github.ref == 'refs/heads/main' || contains(github.ref, 'refs/tags')
     runs-on: ubuntu-22.04
     permissions:
-      contents: read
+      contents: write
```
Comment (pjbgf) [inline .github/workflows/build.yml:49]: Nit: we could add a comment to document why `write` access is needed.

Comment (saschagrunert) [reply .github/workflows/build.yml]: Yep, added a comment to both.

**Review by JAORMX — APPROVED** [review]

**Pull request merged**